`toString()` seems to be broken on 3.6.28

Hi guys, I think the `toString()` method was broken recently, I have a test for mockito-scala that checks a mock `toString` invocation returns its name, but if I try to update to said version it returns the standard `Object` implementation of `toString`.

I tried with the "pure" Java API of Mockito and the same thing happens

Thanks!

Hi @JanMosigItemis 

I was able to reproduce the issue with your test repository. Just replace "mockito-core" with "mockito-inline"

```
            <dependency>
                <groupId>org.mockito</groupId>
                <artifactId>mockito-core</artifactId>
                <version>${mockito.version}</version>
                <scope>test</scope>
            </dependency>
```

```
            <dependency>
                <groupId>org.mockito</groupId>
                <artifactId>mockito-inline</artifactId>
                <version>${mockito.version}</version>
                <scope>test</scope>
            </dependency>
```

You should get:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running de.itemis.jmo.mockito.issue2154.MockitoIssue2154JUnit5Test
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.526 s <<< FAILURE! - in de.itemis.jmo.mockito.issue2154.MockitoIssue2154JUnit5Test
[ERROR] test_stubbed_to_string_works  Time elapsed: 0.054 s  <<< ERROR!
org.mockito.exceptions.misusing.MissingMethodInvocationException: 

when() requires an argument which has to be 'a method call on a mock'.
For example:
    when(mock.getArticles()).thenReturn(articles);

Also, this error might show up because:
1. you stub either of: final/private/equals()/hashCode() methods.
   Those methods *cannot* be stubbed/verified.
   Mocking methods declared on non-public parent classes is not supported.
2. inside when() you don't call method on mock but on some other object.

	at de.itemis.jmo.mockito.issue2154.MockitoIssue2154JUnit5Test.test_stubbed_to_string_works(MockitoIssue2154JUnit5Test.java:26)

[INFO] Running de.itemis.jmo.mockito.issue2154.MockitoIssue2154JUnit4Test
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.041 s <<< FAILURE! - in de.itemis.jmo.mockito.issue2154.MockitoIssue2154JUnit4Test
[ERROR] test_stubbed_to_string_works  Time elapsed: 0.018 s  <<< ERROR!
org.mockito.exceptions.misusing.MissingMethodInvocationException: 

when() requires an argument which has to be 'a method call on a mock'.
For example:
    when(mock.getArticles()).thenReturn(articles);

Also, this error might show up because:
1. you stub either of: final/private/equals()/hashCode() methods.
   Those methods *cannot* be stubbed/verified.
   Mocking methods declared on non-public parent classes is not supported.
2. inside when() you don't call method on mock but on some other object.

	at de.itemis.jmo.mockito.issue2154.MockitoIssue2154JUnit4Test.test_stubbed_to_string_works(MockitoIssue2154JUnit4Test.java:26)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   MockitoIssue2154JUnit4Test.test_stubbed_to_string_works:26 MissingMethodInvocation
[ERROR]   MockitoIssue2154JUnit5Test.test_stubbed_to_string_works:26 MissingMethodInvocation
[INFO] 
[ERROR] Tests run: 4, Failures: 0, Errors: 2, Skipped: 0
```

Let me know if this info is enough for you or if I should fork your repo.

--- a/src/main/java/org/mockito/internal/creation/bytebuddy/InlineBytecodeGenerator.java
+++ b/src/main/java/org/mockito/internal/creation/bytebuddy/InlineBytecodeGenerator.java
@@ -269,9 +269,6 @@ public class InlineBytecodeGenerator implements BytecodeGenerator, ClassFileTran
             throw t;
         }
 
-        // The object type does not ever need instrumentation.
-        targets.remove(Object.class);
-
         if (!targets.isEmpty()) {
             try {
                 assureCanReadMockito(targets);

--- a/src/test/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMakerTest.java
+++ b/src/test/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMakerTest.java
@@ -228,7 +228,8 @@ public class InlineByteBuddyMockMakerTest
         MockSettingsImpl<Object> mockSettings = new MockSettingsImpl<Object>();
         mockSettings.setTypeToMock(Object.class);
         mockSettings.defaultAnswer(new Returns("foo"));
-        Object proxy = mockMaker.createMock(mockSettings, new MockHandlerImpl<Object>(mockSettings));
+        Object proxy =
+                mockMaker.createMock(mockSettings, new MockHandlerImpl<Object>(mockSettings));
 
         assertThat(proxy.toString()).isEqualTo("foo");
     }

--- a/src/test/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMakerTest.java
+++ b/src/test/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMakerTest.java
@@ -220,6 +220,19 @@ public class InlineByteBuddyMockMakerTest
         assertThat(proxy.toString()).isEqualTo("foo");
     }
 
+    /**
+     * @see <a href="https://github.com/mockito/mockito/issues/2154">https://github.com/mockito/mockito/issues/2154</a>
+     */
+    @Test
+    public void should_mock_class_to_string() {
+        MockSettingsImpl<Object> mockSettings = new MockSettingsImpl<Object>();
+        mockSettings.setTypeToMock(Object.class);
+        mockSettings.defaultAnswer(new Returns("foo"));
+        Object proxy = mockMaker.createMock(mockSettings, new MockHandlerImpl<Object>(mockSettings));
+
+        assertThat(proxy.toString()).isEqualTo("foo");
+    }
+
     @Test
     public void should_remove_recursive_self_call_from_stack_trace() throws Exception {
         StackTraceElement[] stack =

